compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2821,7 +2838,11 @@ define_print_and_forward_display! {
     }
 
     ty::AliasTy<'tcx> {
-        p!(print_def_path(self.def_id, self.substs));
+        if self.kind(cx.tcx()) == ty::Inherent {
```

Comment:
It's intentional that I do this check here in the `AliasTy` branch instead of removing `ty::Inherent` from the
match arm `ty::Alias(ty::Projection | ty::Inherent, ref data)` in fn `pretty_print_type` moving it into a new arm
`ty::Alias(ty::Inherent, ref alias_ty)`.

If someone wants to print a bare `AliasTy` that represents an inh proj, then it would just work instead of ICEing.
This case is probably quite rare compared to printing the corresp. `Ty` but let's be robust.